packages/components/src/utils/test/values.js

```diff
@@ -99,3 +99,41 @@ describe( 'isValueNumeric', () => {
 		} );
 	} );
 } );
+
+describe( 'ensureString', () => {
+	it.each( [
+		[ '1', '1' ],
+		[ 'abc', 'abc' ],
+		[ 2e3, '2000' ],
+		[ 42, '42' ],
+		[ -14, '-14' ],
+		[ 0, '0' ],
+		[ '0', '0' ],
+		[ '', '' ],
+		[ NaN, 'NaN' ],
+	] )(
+		'should convert `%s` (unknown) to `%s` (string)',
+		( input, expectedOutput ) => {
+			expect( ensureString( input ) ).toBe( expectedOutput );
+		}
+	);
+} );
+
+describe( 'ensureNumber', () => {
+	it.each( [
+		[ '1', 1 ],
+		[ 'abc', NaN ],
+		[ 2e3, 2000 ],
+		[ 42, 42 ],
+		[ -14, -14 ],
+		[ 0, 0 ],
+		[ '0', 0 ],
+		[ '', NaN ],
+		[ 'NaN', NaN ],
+	] )(
+		'should convert `%s` (unknown) to `%s` (string)',
+		( input, expectedOutput ) => {
+			expect( ensureNumber( input ) ).toBe( expectedOutput );
+		}
+	);
+} );
```

Comment:
Given `null` isn't a `number` would we also need to tweak the name of `ensureNumber` in that case?

Reply:
> Given null isn't a number would we also need to tweak the name of ensureNumber in that case?

Yeah, I could rename that function to `ensureValidNumber` or `ensureFiniteNumber`. Any thoughts?

> The type system will then guide users away from writing (incorrect) code that doesn't handle invalid strings.

I went ahead and changed `ensureNumber` and `ensureString` to return `null` when they are dealing with non-finite numbers (https://github.com/WordPress/gutenberg/pull/45982/commits/eedda2b0aa04f246e176283cc1ab1c38c3d55ca2 and https://github.com/WordPress/gutenberg/pull/45982/commits/5d405b58c72a221771dc55e42236bc70026bccac).

Not sure about the naming, but otherwise how does that feel like?

Reply:
I just realized that these changes break some unit tests, I'll have to look into that too.